src/OpenTelemetry/Metrics/MetricReaderExt.cs

```diff
@@ -294,4 +326,19 @@ private Batch<Metric> GetMetricsBatch()
             return default;
         }
     }
+
+    private void CompleteMetricDeactivation(Metric metric)
+    {
+        lock (metric.StatusLock)
+        {
+            if (metric.Status != Metric.MetricStatusDeactivating)
```

Comment:
Do you need to wrap reads and writes to a `volatile` type under a lock? The reads in `GetMetricsBatch` method don't use this lock.

Reply:
This got removed so it doesn't matter but yes. The reason for that is this is using a double-lock pattern. There is an outer read which is not part of the lock. The volatile is for that outer read to make sure the cpu doesn't cache the value or reorder anything.